README.md

```diff
@@ -1,81 +1,92 @@
-2. Expose documentation/educational information about data breaches in the Firefox UI - for example, a "Learn more" link in the notification mentioned above leading to a support page.
-3. Offer a way for interested users to learn about and opt into a service that notifies them (e.g. via email) when they may be affected by breaches in the future.
+* [node](https://nodejs.org/) (with npm)
 
-## Privacy Concerns
+### Install
 
-The third goal brings up some privacy concerns, since users would need to supply an email address to receive notifications. Who is the custodian of this data? Can we avoid sending user data to haveibeenpwned.com? Can we still offer useful functionality to users who opt out of subscribing their email address? While the project is still in infancy, the idea is to offer as much utility as possible while respecting the user's privacy.
+1. clone and change to the directory:
 
-## Try it (basic testing instructions for Unix OSes as of Nov 23 2017)
+        git clone https://github.com/nhnt11/blurts-server.git
+        cd blurts-server
 
-### Clone
+2. Install dependencies:
 
-```
-$ git clone https://github.com/nhnt11/BreachAlerts.git
-```
+        npm install
 
-### Server (to test subscribing an email address and sending a "Breach Alert" email)
+3. Copy the `.env-dist` file to `.env`:
 
-Prerequisites: Node.js and npm (example commands assume you have `node` and `npm` in your path)
+        cp .env-dist .env
 
-You'll need to supply SMTP credentials for sending the email. To do so, create a file `smtp-credentials.json` in the `BreachAlerts/server` folder:
+### Run
 
-```
-{
-  "username": "<username>",
-  "password": "<password>"
-}
+1. Source the `.env` file to set required environment variables:
 
-```
+        source .env
+
+2. Run the server:
 
-NOTE: Currently the Gmail SMTP server is hardcoded, so you'll need to modify that in server.js if you're using something else.
+        node server.js
 
+#### Emails
+
+The included `.env-dist` sets `DEBUG_DUMMY_SMTP=1` which disables emails.
+
+To send emails, you'll need to unset `DEBUG_DUMMY_SMTP` and supply real SMTP
+config values for sending email.
+
+You can set and source these via the `.env` file, or set them directly:
 ```
-$ cd BreachAlerts/server
-$ npm install
-$ node server.js
+export DEBUG_DUMMY_SMTP=0
+export SMTP_HOST=<your-smtp-host>
+export SMTP_PORT=<your-smtp-port>
+export SMTP_USERNAME=<your-username>
+export SMTP_PASSWORD=<your-password
 ```
 
-### Client:
+#### Firefox Accounts
 
-Prerequisites: currently Firefox 58.0 or 59.0.
+To use Firefox Accounts, you'll need to [create an FxA Oauth Client](https://oauth-stable.dev.lcip.org/console/clients)
+and then set some `OAUTH` config values.
 
+You can set and source these via the `.env` file, or set them directly:
 ```
-$ cd BreachAlerts/client-addon
-$ ./package.sh # This creates a client-addon.xpi file in the dist/ folder
+OAUTH_CLIENT_ID=<your-fxa-oauth-client-id>
```

Comment:
We can probably mention the existing dev OAuth relier account. I found precedent for exposing staging/development credentials elsewhere (can't remember now, will try to find it), so I put our client_id and client_secret in .env_dist.

Reply:
Is that client configured to redirect back to localhost:6060 properly? When I used the values in `.env-dist`, I got "invalid request parameter" at http://localhost:6060/oauth/redirect?state=2e2...6d4&code=972...8f5&action=signin when I signed in with FxA.

Reply:
It should "just work". The FxA local dev console specifically says that it allows localhost redirects for ease of development. Thanks though, I'll look into this.